File: Week 1/3.py
class Solution:
    def getSecondLargest(self, arr):
        largest = max(arr)
        second_largest = -1  # Start with -1, as required

        for a in arr:
            if a != largest:
                if second_largest == -1 or a > second_largest:
                    second_largest = a

        return second_largest

if __name__ == "__main__":
    # Taking array input from user
    arr_input = input("Enter the array elements separated by spaces: ")
    arr = list(map(int, arr_input.strip().split()))

    # Creating object and calling the function
    solution = Solution()
    result = solution.getSecondLargest(arr)

    # Printing result
    print("Second largest element:", result)

File: Week 1/3.c
#include <stdio.h>

int getSecondLargest(int *arr, int n) {
    int largest = -1, second_largest = -1;

    for (int i = 0; i < n; i++) {
        if (arr[i] > largest)
            largest = arr[i];
    }

    for (int i = 0; i < n; i++) {
        if (arr[i] != largest && arr[i] > second_largest)
            second_largest = arr[i];
    }

    return second_largest;
}

int main() {
    int n;

    printf("Enter the number of elements: ");
    scanf("%d", &n);

    int arr[n];
    printf("Enter %d elements:\n", n);
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    int result = getSecondLargest(arr, n);
    printf("Second largest element: %d\n", result);

    return 0;
}
